drivers/adc/ad717x/ad717x.c

```diff
@@ -812,8 +845,7 @@ int32_t AD717X_Init(ad717x_dev **device,
 	dev->active_device = init_param.active_device;
 	dev->num_channels = init_param.num_channels;
 
-	for (setup_index = 0; setup_index < NO_OS_ARRAY_SIZE(init_param.setups);
-	     setup_index++) {
+	for (setup_index = 0; setup_index < init_param.num_setups; setup_index++) {
```

Comment:
Question: is the number of setups added so that we can initialize fewer setups than maximum?

Reply:
Yes, num_setups has been added as a member in the `ad717x_init_param` structure.